### ISSUE TITLE
Strange error message for code that combines mutual and abstract

### ISSUE THREAD

**Issue by nad** [agda/agda#3742]
Code:
```agda
module Bug where

record R (A : Set) (B : A → Set₁) : Set₁ where

mutual

  abstract

    data D (A : Set) : Set where
      c : D A

  F : Set → Set₁
  F A = R _ λ (_ : D A) → Set
```
Error message:
```
Not in scope:
  Bug.D.c
  at […]/Bug.agda:10,7-8
    (did you mean 'c'?)
when checking that the expression λ (_ : D A) → Set has type
_A_3 → Set₁
```
The code is accepted if the first underscore is replaced by `D A`.

**Comment by andreasabel** [comment]
This worked fine until including 2.5.1.1.  I am starting agda-bisect...

**Comment by andreasabel** [comment]
Hours later, fighting with a cabal bug https://github.com/haskell/cabal/issues/6031, it seems like the culprit is commit a1bb2cfe7c4eed49666b1cb4978a5a634e8bb84f
```
  An internal error has occurred. Please report this as a bug.
  Location of the error: src/full/Agda/TypeChecking/MetaVars/Occurs.hs:362
Verdict: Bad
```

- Fixed #2250, fixed #2248 by clarifying API of getConstInfo to return AbstractDefn instead of Axiom for abstract definitions

**Comment by andreasabel** [comment]
Fixed by adding some `ignoreAbstractMode` in `metaOccurs` check, which looks into the types of (even abstract) constructors of mutual data types for meta variable occurrences.
This fix can go into 2.6.0.1.
